app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -33,6 +33,38 @@ cat << EOF > $SSL_CA_CERT
 <%= foreman_server_ca_cert %>
 EOF
 
+<% unless @repo.blank? -%>
+echo '#'
+echo '# Adding repository'
+echo '#'
+
+if [ x$ID = xrhel ] || [ x$ID = xfedora ] || test "${ID_LIKE#*rhel*}" != "$ID_LIKE" ; then
+  cat << EOF > /etc/yum.repos.d/foreman_registration.repo
+[foreman_register]
+name=foreman_register
+baseurl=<%= @repo %>
+enabled=1
+gpgcheck=<%= @repo_gpg_key_url.present? ? 1 : 0 -%>
+gpgkey=<%= @repo_gpg_key_url %>
```

Comment:
since it was changed to the URL, I think we need to download the key first to some file and then use `file://path/to/foreman_registration.key`, now I'm getting the URL here, which would not work

```
enabled=1
gpgcheck=1gpgkey=http://yum.theforeman.org/RPM-GPG-KEY-foreman
EOF
```

so it should do something like
```
curl <%= @repo_gpg_key_url %> > /some/temp_file
rpm --import /some/temp_file
```

you already do something similar with `apt-key add` below

Reply:
Yum understands HTTP(S) URLs for GPG keys just fine. I don't think we should worry about it.

Reply:
It should work, I tried that and it worked without any problems.

Reply:
ok, fine by me